src/code/z_player_lib.c

```diff
@@ -2229,38 +2966,397 @@ struct_80128388_arg1 D_801C0C54[] = {
+    { 80, 80, 255 },   // PLAYER_BOTTLE_FAIRY            // PLAYER_IA_BOTTLE_FAIRY
 };
 
 Vec3f D_801C0CE8[PLAYER_FORM_MAX] = {
-    { 0.0f, 0.0f, 0.0f },    { 300.0f, 300.0f, -230.0f }, { 0.0f, 90.0f, -50.0f },
-    { 0.0f, 20.0f, -60.0f }, { 0.0f, 0.0f, 0.0f },
+    { 0.0f, 0.0f, 0.0f },        // PLAYER_FORM_FIERCE_DEITY
+    { 300.0f, 300.0f, -230.0f }, // PLAYER_FORM_GORON
+    { 0.0f, 90.0f, -50.0f },     // PLAYER_FORM_ZORA
+    { 0.0f, 20.0f, -60.0f },     // PLAYER_FORM_DEKU
+    { 0.0f, 0.0f, 0.0f },        // PLAYER_FORM_HUMAN
 };
-Vec3f D_801C0D24[PLAYER_FORM_MAX] = {
-    { 200.0f, 300.0f, 0.0f }, { 200.0f, 200.0f, 0.0f }, { 200.0f, 300.0f, 0.0f },
-    { 200.0f, 150.0f, 0.0f }, { 200.0f, 200.0f, 0.0f },
+
+void Player_DrawBunnyHood(PlayState* play) {
+    Mtx* temp_a1 = GRAPH_ALLOC(play->state.gfxCtx, 2 * sizeof(Mtx));
+    Vec3s sp2C;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gSPSegment(POLY_OPA_DISP++, 0x0B, temp_a1);
+
+    Matrix_Push();
+
+    sp2C.x = sBunnyEarKinematics.rot.y + 0x3E2;
+    sp2C.y = sBunnyEarKinematics.rot.z + 0xDBE;
+    sp2C.z = sBunnyEarKinematics.rot.x - 0x348A;
+    Matrix_SetTranslateRotateYXZ(97.0f, -1203.0f, -240.0f, &sp2C);
+
+    Matrix_ToMtx(temp_a1++);
+
+    sp2C.x = sBunnyEarKinematics.rot.y - 0x3E2;
+    sp2C.y = -sBunnyEarKinematics.rot.z - 0xDBE;
+    sp2C.z = sBunnyEarKinematics.rot.x - 0x348A;
+    Matrix_SetTranslateRotateYXZ(97.0f, -1203.0f, 240.0f, &sp2C);
+
+    Matrix_ToMtx(temp_a1);
+
+    Matrix_Pop();
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void func_80127B64(struct_801F58B0 arg0[], s32 count, Vec3f* arg2) {
+    s32 i;
+
+    for (i = 0; i < count; i++, arg0++) {
+        Math_Vec3f_Copy(&arg0->unk_00, arg2);
+        Math_Vec3f_Copy(&arg0->unk_0C, &gZeroVec3f);
+        arg0->unk_18 = 0;
+        arg0->unk_1A = 0;
+    }
+}
+
+void func_80127BE8(PlayState* play, Vec3f* arg1) {
+    Vec3f sp2C;
+    f32 sp28;
+
+    D_801C0BE0.y = Rand_ZeroFloat(0.07f) + -0.1f;
+    D_801C0BEC.y = Rand_ZeroFloat(0.1f) + 0.04f;
+    if (Rand_ZeroOne() < 0.5f) {
+        sp28 = -1.0f;
+    } else {
+        sp28 = 1.0f;
+    }
+
+    D_801C0BE0.x = (Rand_ZeroFloat(0.2f) + 0.1f) * sp28;
+    D_801C0BEC.x = 0.1f * sp28;
+    if (Rand_ZeroOne() < 0.5f) {
+        sp28 = -1.0f;
+    } else {
+        sp28 = 1.0f;
+    }
+
+    D_801C0BE0.z = (Rand_ZeroFloat(0.2f) + 0.1f) * sp28;
+    D_801C0BEC.z = 0.1f * sp28;
+    sp2C.x = arg1->x;
+    sp2C.y = Rand_ZeroFloat(15.0f) + arg1->y;
+    sp2C.z = arg1->z;
+    EffectSsKirakira_SpawnDispersed(play, &sp2C, &D_801C0BE0, &D_801C0BEC, &D_801C0BF8, &D_801C0BFC, -50, 11);
+}
+
+void func_80127DA4(PlayState* play, struct_801F58B0 arg1[], struct_80128388_arg1 arg2[], s32 arg3, Vec3f* arg4,
+                   Vec3f* arg5, u32* arg6) {
+    struct_801F58B0* phi_s1 = &arg1[1];
+    Vec3f spB0;
+    Vec3f spA4;
+    f32 f22;
+    f32 f28;
+    f32 f24;
+    f32 f20;
+    f32 f0;
+    f32 sp8C = -1.0f;
+    s32 i;
+    s16 s0;
+    s16 s2;
```

Comment:
These temps feel too abbreviated, but since it's temp anyway, maybe it's fine

Reply:
Presumably this will be clearer once the ACTORCTX_FLAG stuff is sorted out?

Reply:
We already know `ACTORCTX_FLAG_3` is what triggers the great fairy mask to start releasing pink particles, which is the indicator that there is an unfound stray fairy in the room:
https://discord.com/channels/688807550715560050/873211000122384415/1062553911493398630